### DISCUSSION TITLE
How to observe current screen in backstack?

### DISCUSSION THREAD

**Discussion by berhram** [bumble-tech/appyx#199] [Q&A] [answered]
For example, in my `RootNode` I have `Scaffold` with bottom and top bar. How can I observe what screen in active and recompose my `Scaffold` each time when screen changed. I tried `val scaffoldState = remember(backstack.activeElement) { ... }` but it doesn't work (`remember` doesn't trigger)?

**Comment by KovalevAndrey** [comment, marked as answer]
Hi @berhram you can implement your own extension that transforms `BackStack` element into a `State` like this:

```
val <T : Any> BackStack<T>.activeElementState: State<T?>
    @Composable
    get() = elements.map { it.activeElement }.collectAsState(initial = null)
```        

and your client code would look like this:

```
val activeElement by backStack.activeElementState

LaunchedEffect(activeElement) {
    // do something with your active element
}
```        

In the meantime we can add support of this to the library

**Reply by berhram** [reply]
Thanks!
